### PR TITLE
issue-448 :Convert Pytest-CVP Power Supply Status case to Vane

### PR DESCRIPTION
**Pull request by Ashutosh-GSLAB**
# Please include a summary of the changes

nrfu_tests/test_definition.yaml: Updated test def with testcase NRFU6.9
nrfu_tests/test_definition.yaml.j2: Updated with test case NRFU6.9
nrfu_tests/test_system_hardware_power_supply_status.py Added test case for NRFU6.9

# Any specific logic/part of code you need extra attention on

This testcase will verify the power supply  status in 'show system environment power' command output. If power supply status are not found test case will fail with assertion error.

# Include the Issue number and link

https://github.com/aristanetworks/vane/issues/448

# List/Attach any dependencies/past issues that are required for this change/provide context

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [ ] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [x] Other (NRFU testcases)

# Effort required on reviewers end

- - [ ] Easy
- - [x] Medium
- - [ ] Hard 

# How Has This Been Tested?
1. Testcase pass when power supplies is 'Ok'.  ( pass without j2 )
  
[report.html_2309141445.zip](https://github.com/aristanetworks/vane/files/12607103/report.html_2309141445.zip)

[report_2309141445.docx](https://github.com/aristanetworks/vane/files/12607105/report_2309141445.docx)

2. Testcase pass when power supplies is 'Ok'.  ( pass with j2 )
  
[report.html_2309141447_with_j2.zip](https://github.com/aristanetworks/vane/files/12607120/report.html_2309141447_with_j2.zip)

[report_2309141448.docx](https://github.com/aristanetworks/vane/files/12607110/report_2309141448.docx)

Fail with hardcoded output 

3. Testcase fail when power supplies is 'not Ok'. 

[report.html_2309141457.zip](https://github.com/aristanetworks/vane/files/12607212/report.html_2309141457.zip)

[report_2309141457.docx](https://github.com/aristanetworks/vane/files/12607215/report_2309141457.docx)

4.  assert fail

[report.html_2309141459.zip](https://github.com/aristanetworks/vane/files/12607232/report.html_2309141459.zip)

[report_2309141500.docx](https://github.com/aristanetworks/vane/files/12607234/report_2309141500.docx)

latest_report:

[report.html_2309181401.zip](https://github.com/aristanetworks/vane/files/12645901/report.html_2309181401.zip)

[report_2309181401.docx](https://github.com/aristanetworks/vane/files/12645905/report_2309181401.docx)

## Bug fix

    Include before and after snapshots/screenshots/changed logs
    
## New feature

    Ensure current test cases pass and include newer test cases if required
    
# CI pipeline result

- - [ ] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update

    If applicable, ensure documentation has been updated for ReadMe, Getting Started guide, and Style guide
    
# Additional comments

### CODE REVIEW

File: nrfu_tests/test_definition.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+- name: nrfu_tests
+  testcases:
+  - name: test_system_hardware_power_supply_status
+    description: Test case for verification of system power supply status on the device.
```
Comment (hsisalgslab) [inline nrfu_tests/test_definition.yaml:4]:
```suggestion
    description: Test case for verification of system hardware power supply status.
```

File: nrfu_tests/test_definition.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    test_id: NRFU6.9
+    show_cmd: show system environment power
+    expected_output: null # Expected output is formed inside a test case file dynamically
+    test_criteria: Status of all power supplies should be 'Ok'. # Setting test case’s pass / fail criteria for reporting
+    report_style: modern # Setting report_style as modern If report type is set to modern, then it will update steps, assumptions, and external systems in the docx report.
+    criteria: names
+    filter:
+    - blb1-fab01
```
Comment (hsisalgslab) [inline nrfu_tests/test_definition.yaml:12]:
```suggestion
    - BLFE1
```

File: nrfu_tests/test_system_hardware_power_supply_status.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+from vane.config import dut_objs, test_defs
+from vane import tests_tools
+
+
+TEST_SUITE = "nrfu_tests"
+
+
+@pytest.mark.nrfu_test
+@pytest.mark.system
+class SystemHardwarePowerSupplyTests:
```
Comment (hsisalgslab) [inline nrfu_tests/test_system_hardware_power_supply_status.py:20]: https://github.com/aristanetworks/vane/pull/489
update location of directory structure  as per this MR  for all files .

File: nrfu_tests/test_system_hardware_power_supply_status.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+            for power_supply, power_supply_info in power_supplies.items():
+                tops.expected_output["power_supplies"].update({power_supply: {"status": "ok"}})
+                tops.actual_output["power_supplies"].update(
+                    {power_supply: {"status": power_supply_info["state"]}}
+                )
+
+            # Forming output message in case of test case failure
+            if tops.actual_output != tops.expected_output:
+                unhealthy_power_supplies = []
```
Comment (hsisalgslab) [inline nrfu_tests/test_system_hardware_power_supply_status.py:89]:
```suggestion
                bad_power_supplies= []

```

File: nrfu_tests/test_system_hardware_power_supply_status.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            # Skipping testcase if device is vEOS.
+            if "vEOS" in version_output.get("modelName"):
+                pytest.skip(f"{tops.dut_name} is vEOS device, hence test skipped.")
+
+            """
+            TS: Running `show system environment power` command on the device and
+            verifying status for all power supplies should be 'OK'.
+            """
+            power_supplies_output = dut["output"][tops.show_cmd]["json"]
+            power_supplies_output = {"powerSupplies": {}}
```
Comment (hsisalgslab) [inline nrfu_tests/test_system_hardware_power_supply_status.py:69]: do we need this line ? test case will always fail with assertion

File: nrfu_tests/test_system_hardware_power_supply_status.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        self.output = ""
+        tops.actual_output = {"power_supplies": {}}
+        tops.expected_output = {"power_supplies": {}}
+
+        # Output message if test result is passed
+        tops.output_msg = "Status of all power supplies are 'Ok'"
+
+        try:
+            """
+            TS: Running 'show version' command on device and Skipping the test case
```
Comment (hsisalgslab) [inline nrfu_tests/test_system_hardware_power_supply_status.py:49]:
```suggestion
            TS: Running `show version` command on device and skipping the test case

```

**Review by hsisalgslab — APPROVED** [review]

File: sample_network_tests/nrfu/test_definition.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+- name: sample_network_tests
+  testcases:
+  - name: test_system_hardware_power_supply_status
+    description: Test case for verification of system hardware power supply status.
```
Comment (geetanjalimanegslab) [inline sample_network_tests/nrfu/test_definition.yaml:4]:
```suggestion
    description: Test case for the verification of system hardware power supply status.
```

File: sample_network_tests/nrfu/test_definition.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+    test_id: NRFU6.9
+    show_cmd: show system environment power
+    expected_output: null # Expected output is formed inside a test case file dynamically
+    test_criteria: Status of all power supplies should be 'Ok'. # Setting test case’s pass / fail criteria for reporting
```
Comment (geetanjalimanegslab) [inline sample_network_tests/nrfu/test_definition.yaml:8]:
```suggestion
    test_criteria: Status of all power supplies should be 'Ok'. # Setting test case’s pass/fail criteria for reporting
```

File: sample_network_tests/nrfu/test_definition.yaml [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+    report_style: modern # Setting report_style as modern If report type is set to modern, then it will update steps, assumptions, and external systems in the docx report.
+    criteria: names
+    filter:
+    - BLFE1
```
Comment (geetanjalimanegslab) [inline sample_network_tests/nrfu/test_definition.yaml:12]: fix the indentation for filter

File: sample_network_tests/nrfu/test_definition.yaml.j2 [hunk 10]
```diff
@@ -0,0 +1,11 @@
+- name: sample_network_tests
+  testcases:
+    - name: test_system_hardware_power_supply_status
+      description: Test case for verification of system hardware power supply status.
+      test_id: NRFU6.9
+      show_cmd: show system environment power
+      expected_output: null
+      test_criteria: Status of all power supplies should be 'Ok'. # Setting test case's pass / fail criteria for reporting
```
Comment (geetanjalimanegslab) [inline sample_network_tests/nrfu/test_definition.yaml.j2:8]:
```suggestion
      test_criteria: Status of all power supplies should be 'Ok'. # Setting test case's pass/fail criteria for reporting

```

File: sample_network_tests/nrfu/test_system_hardware_power_supply_status.py [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+            tests_definitions(dict): test suite and test case parameters.
+        """
+
+        tops = tests_tools.TestOps(tests_definitions, TEST_SUITE, dut)
+        self.output = ""
+        tops.actual_output = {"power_supplies": {}}
+        tops.expected_output = {"power_supplies": {}}
+
+        # Output message if test result is passed
+        tops.output_msg = "Status of all power supplies are 'Ok'"
```
Comment (geetanjalimanegslab) [inline sample_network_tests/nrfu/test_system_hardware_power_supply_status.py:45]:
```suggestion
        tops.output_msg = "Status of all the power supplies is 'Ok'"

```

File: sample_network_tests/nrfu/test_system_hardware_power_supply_status.py [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+                for power_supply, expected_state in tops.expected_output["power_supplies"].items():
+                    actual_status = tops.actual_output["power_supplies"][power_supply]
+                    if actual_status == expected_state:
+                        continue
+                    bad_power_supplies.append({power_supply: actual_status.get('status')})
+
+                if bad_power_supplies:
+                    power_supply_status = "\n".join(bad_power_supplies)
+                    tops.output_msg = (
+                        "For following power supplies status is not found as 'OK'. current"
```
Comment (geetanjalimanegslab) [inline sample_network_tests/nrfu/test_system_hardware_power_supply_status.py:97]:
```suggestion
                        "For the following power supplies status is not found as 'OK'. current"

```

**Review by geetanjalimanegslab — APPROVED** [review]

**Review by kgrozis-arista — APPROVED** [review]

**Pull request merged**